test/functional/tests/volumes/test_volume_partitions.py

```diff
@@ -0,0 +1,182 @@
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.parametrize("partition_table", PartitionTable)
+@pytest.mark.parametrize("filesystem", Filesystem)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_partition_create_cas(partition_table, filesystem, cache_mode):
+    """
+        title: Volume test for preserving partition table on core device after stopping OpenCAS.
+        description: |
+          Validation of the ability of CAS to preserve partition table on core device
+          after stopping cache.
+        pass_criteria:
+          - Md5 sums shall on partitions be identical before and after running cache.
+          - Partition table shall be preserved on core device.
+    """
+    with TestRun.step(f"Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(256, Unit.MebiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+
+    with TestRun.step(f"Start cache."):
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Create partitions on exported device."):
+        core_size = []
+        for i in range(cores_number):
+            core_size.append(Size(1, Unit.GibiByte))
+        core.block_size = core_dev.block_size
+        disk_utils.create_partitions(core, core_size, partition_table)
+
+    with TestRun.step("Create filesystem on core devices."):
+        for part in core.partitions:
+            part.create_filesystem(filesystem)
+
+    with TestRun.step("Mount core devices and create test files."):
+        counter = 1  # There is one partition (4th) which is omitted while creating cache
+        files = []   # and continuous numbering is needed here.
+        for part in core.partitions:
+            mount_path = mount_point + str(counter)
+            part.mount(mount_path)
+            test_file_path = f"{mount_path}/test_file"
+            files.append(fs_utils.create_test_file(test_file_path))
+            counter += 1
+
+    with TestRun.step("Check md5 sums of test files."):
+        test_files_md5sums = []
+        for file in files:
+            test_files_md5sums.append(file.md5sum())
+
+    with TestRun.step("Unmount core devices."):
+        for part in core.partitions:
+            part.unmount()
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+    with TestRun.step("Read partitions on core device."):
+        for part in core.partitions:
+            part.parent_device = core_dev
+            new_part = Partition(part.parent_device, part.type, part.number)
+            core_dev.partitions.append(new_part)
+
+    with TestRun.step("Mount core devices."):
+        counter = 1
+        for core in core_dev.partitions:
+            mount_path = mount_point + str(counter)
```

Comment:
you could (but don't have to) use: f'{mount_point}{counter}'

Reply:
Yeah, Your method is more readable. Thanks!